diff -urNp -x '*.orig' TK-4_13_0-20161107.org/src/Makefile TK-4_13_0-20161107/src/Makefile
--- TK-4_13_0-20161107.org/src/Makefile	2016-01-25 03:47:52.000000000 +0100
+++ TK-4_13_0-20161107/src/Makefile	2021-09-28 23:27:25.531460395 +0200
@@ -56,6 +56,7 @@ OSPOBJS = osppkcs1.o osppkcs8.o osppkcs7
           ospciscoext.o ospcapind.o ospcapcnf.o ospstir.o $(SSLOBJ)
           
 OSPLIB = $(LIBDIR)/libosptk.a
+OSPLIBSO = $(LIBDIR)/libosptk.so.0.0.0
 
 .SUFFIXES: .o .c 
 
@@ -72,14 +73,20 @@ default:
 
 build: $(OSPLIB)
 
-$(OSPLIB): $(OSPOBJS)
+$(OSPLIB): $(OSPOBJS) $(OSPLIBSO)
 	$(AR) -r $(OSPLIB) $(OSPOBJS)
 
+$(OSPLIBSO): $(OSPOBJS)
+	$(CC) -o $(OSPLIBSO) $(LDFLAGS) -Wl,-soname -Wl,libosptk.so.0 $(OSPOBJS) -lssl -lm -shared -lcrypto -lpthread
+
 install: build
 	install -d $(INSTALL_PATH)/include/osp
 	cp $(INCDIR)/osp/*.h $(INSTALL_PATH)/include/osp/
 	install -d $(INSTALL_PATH)/lib
 	cp $(OSPLIB) $(INSTALL_PATH)/lib/
+	cp $(OSPLIBSO) $(INSTALL_PATH)/lib/
+	ln -s libosptk.so.0.0.0 $(INSTALL_PATH)/lib/libosptk.so.0
+	ln -s libosptk.so.0.0.0 $(INSTALL_PATH)/lib/libosptk.so
 
 clean:
 	rm -f $(OSPLIB) $(OSPOBJS)
